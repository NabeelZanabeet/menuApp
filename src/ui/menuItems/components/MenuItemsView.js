import * as React from 'react';
import { Image, Text, View, TouchableOpacity, ScrollView } from 'react-native';
import styles from './MenuItemsView.styles';
import { backIcon, loader } from '../../../assets/images';

const MenuItemsView = ({ isProcessing, items, onPressBackButton }) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={styles.backTouchable}
        onPress={onPressBackButton}>
        <Image style={styles.backIcon} source={backIcon} resizeMode="contain" />
      </TouchableOpacity>
      <Text style={styles.title}>Items</Text>
      {items && !isProcessing && (
        <ScrollView
          style={styles.scrollView}
          showsVerticalScrollIndicator={false}>
          {items.map((item) => (
            <View key={item.id} style={styles.itemCard}>
              <Text style={styles.name}>{`Name: ${item.name}`}</Text>
              <Text style={styles.price}>{`Price: ${item.price}`}</Text>
            </View>
          ))}
        </ScrollView>
      )}
      {!items && isProcessing && (
        <Image style={styles.loader} source={loader} resizeMode="contain" />
      )}
    </View>
  );
};

export default MenuItemsView;
