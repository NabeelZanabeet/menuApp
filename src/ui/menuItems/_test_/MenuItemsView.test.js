import 'react-native';
import React from 'react';
import deepRenderer from 'react-test-renderer';
import MenuItemsView from '../components/MenuItemsView';

describe('MenuItemsView component', () => {
  test('renders correctly', () => {
    const tree = deepRenderer.create(<MenuItemsView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
