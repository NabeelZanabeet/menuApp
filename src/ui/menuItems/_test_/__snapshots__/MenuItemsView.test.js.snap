// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemsView component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 24,
        "opacity": 1,
        "position": "absolute",
        "top": 24,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/backIcon.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 17,
        "position": "absolute",
        "textAlign": "center",
        "top": 24,
      }
    }
  >
    Items
  </Text>
</View>
`;
